src/providers/list/components/InventoryCell.tsx

```diff
@@ -0,0 +1,41 @@
+import type { FC } from 'react';
+import { getResourceFieldValue } from 'src/components/common/FilterGroup/matchers';
+import { TableEmptyCell } from 'src/modules/Providers/utils/components/TableCell/TableEmptyCell';
+import { TableIconCell } from 'src/modules/Providers/utils/components/TableCell/TableIconCell';
+import type { CellProps } from 'src/modules/Providers/views/list/components/CellProps';
+
+import { PROVIDER_TYPES, ProvidersTableResourceFieldId } from '../utils/constants';
+
+import { OpenshiftNetworkCell } from './OpenshiftNetworkCell';
+import { VSphereHostCell } from './VSphereHostCell';
+
+type InventoryCellProps = {
+  icon?: JSX.Element;
+} & CellProps;
+
+export const InventoryCell: FC<InventoryCellProps> = ({ data, fieldId, fields, icon }) => {
+  const { inventory, provider } = data;
+  const type = provider?.spec?.type;
+
+  const value = getResourceFieldValue({ ...provider, inventory }, fieldId, fields);
+
+  if (value === undefined) {
+    return <TableEmptyCell />;
+  }
+
+  if (
+    (type as keyof typeof PROVIDER_TYPES) === PROVIDER_TYPES.openshift &&
```

Comment:
nit: Maybe just cast this where the constant is defined above since it's cast in the same way on line 34.